### ISSUE TITLE
Rubocop warns about the deprecation of `RuboCop::Cop::EnforceSuperclass` even after recommended Rubop Rails version upgrade

### ISSUE THREAD

**Issue by norman** [rubocop/rubocop#14094]
Hello,

This is not a big problem for me and I assume this is also a low priority for you, but I'm reporting so you can deal with it when you decide the time is right.

Rubocop warns about the deprecation of `RuboCop::Cop::EnforceSuperclass` coming in Rubcop 2.0. The warning says to upgrade to RuboCop Rails 2.9 of newer to continue.  But, I am running Rubocop Rails 2.31.0 but still see this warning on every spec run. This is a minor annoyance; I would love to not see this message all the time since I already did what was needed to make it go away.

--------

## Expected behavior

Running Rubocop in conjunction with Rubocop Rails 2.9.0 and above should not warn about the deprecation of `      RuboCop::Cop::EnforceSuperclass`.

## Actual behavior

Running Rubocop in conjuntion with Rubocop Rails 2.31.0 still warns about the deprecation with the following message in yellow:

> `RuboCop::Cop::EnforceSuperclass` is deprecated and will be removed in RuboCop 2.0. Please upgrade to RuboCop Rails 2.9 or newer to continue.

Note that Rubocop Rails 2.31.0 (and I assume all versions after 2.9) include their own version of `RuboCop::Cop::EnforceSuperclass`.

## Steps to reproduce the problem

Defines the following custom cop:

```ruby
module MyApp
  module Cops
    class ApplicationDecorator < ::RuboCop::Cop::Base
      extend RuboCop::Cop::AutoCorrector

      MSG = "Decorators should subclass `ApplicationDecorator`."
      SUPERCLASS = "ApplicationDecorator"
      BASE_PATTERN = "(const (const {nil? cbase} :Draper) :Decorator)"

      include RuboCop::Cop::EnforceSuperclass
    end
  end
end
```

## RuboCop version

```
1.75.2 (using Parser 3.3.7.4, Prism 1.4.0, rubocop-ast 1.44.0, analyzing as Ruby 3.3, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-rails 2.31.0
  - rubocop-rspec 3.5.0
  - rubocop-thread_safety 0.7.2
  - rubocop-rspec_rails 2.31.0
  - rubocop-capybara 2.22.1
  - rubocop-performance 1.25.0
```

**Comment by Earlopain** [comment]
Hi! Without looking to much into it, I'd guess that `rubocop-rails` hasn't been required yet at that point. Doing so should replace it with something that doesn't emit a warning.

**Comment by norman** [comment]
> Hi! Without looking to much into it, I'd guess that `rubocop-rails` hasn't been required yet at that point. Doing so should replace it with something that doesn't emit a warning.

Thanks. Requiring `require "rubocop/cop/rails_cops"` does make the deprecation warning go away.

I'll leave the issue open for the maintainers to triage in case there's any other action they want to take.
